src/raft/replication.c

```diff
@@ -1468,12 +1506,21 @@ int replicationInstallSnapshot(struct raft *r,
 /* Apply a RAFT_COMMAND entry that has been committed. */
 static int applyCommand(struct raft *r,
 			const raft_index index,
-			const struct raft_buffer *buf)
+			const raft_term term,
+			const struct raft_buffer *buf,
+			struct raft_entry_local_data ld,
+			bool is_local)
 {
 	struct raft_apply *req;
 	void *result;
 	int rv;
-	rv = r->fsm->apply(r->fsm, buf, &result);
+
+	if (r->fsm->version >= 4 && r->fsm->apply2 != NULL) {
+		bool is_mine = is_local && term == r->current_term;
```

Comment:
I think this should be part of the documentation of the apply2 function, stating what it means for an entry to be mine. Also, let's try to think about a better name that conveys that it has to have been created by the node while it was the leader (if we cannot, then the documentation is enough and we can keep it as `is_mine`). For example, what about `created_as_leader_current_term`? (maybe too verbose)

Reply:
Agreed about documentation. Maybe a more verbose name in the header file and the concise `is_mine` when it's referred to in `.c`?

Reply:
That would work.